File: src-recovered/bron-main/scratch/restore_components.js

const fs = require('fs');
const path = require('path');

const logPath = 'C:\\Users\\devev\\.gemini\\antigravity\\brain\\a95e4d70-ca8f-462a-9b43-ef09835c11a1\\.system_generated\\logs\\overview.txt';
const outputDir = 'd:\\C_Drive_Transfer\\bron\\src\\renderer\\components';

if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir, { recursive: true });
}

const components = [
    'BrowserToolbar.tsx',
    'TabBar.tsx',
    'AgentSidebar.tsx',
    'MemoryPanel.tsx',
    'SettingsPanel.tsx'
];

const filesFound = {};

const content = fs.readFileSync(logPath, 'utf8');
const lines = content.split('\n');

lines.forEach(line => {
    try {
        const data = JSON.parse(line);
        if (data.tool_calls) {
            data.tool_calls.forEach(call => {
                if (['write_to_file', 'replace_file_content'].includes(call.name)) {
                    const args = call.args;
                    const target = args.TargetFile || '';
                    components.forEach(comp => {
                        if (target.includes(comp)) {
                            // We use regex to find the content because the JSON parsing might have escaped it differently
                            // But since we parsed the line as JSON, args.CodeContent or args.ReplacementContent should be strings
                            let fileContent = args.CodeContent || args.ReplacementContent;
                            if (fileContent) {
                                // Strip outer quotes if they exist and unescape
                                if (fileContent.startsWith('"') && fileContent.endsWith('"')) {
                                    fileContent = fileContent.slice(1, -1);
                                }
                                // The string from JSON.parse is already unescaped for standard characters,
                                // but the logs might have double-escaped newlines like "\\n"
                                fileContent = fileContent.replace(/\\n/g, '\n').replace(/\\"/g, '"').replace(/\\\\/g, '\\');
                                filesFound[comp] = fileContent;
                            }
                        }
                    });
                }
            });
        }
    } catch (e) {
        // Ignore lines that aren't JSON
    }
});

Object.keys(filesFound).forEach(comp => {
    const destPath = path.join(outputDir, comp);
    fs.writeFileSync(destPath, filesFound[comp], 'utf8');
    console.log(`Restored ${comp}`);
});

File: scratch/restore_components.py

import json
import os
import re

log_path = r"C:\Users\devev\.gemini\antigravity\brain\a95e4d70-ca8f-462a-9b43-ef09835c11a1\.system_generated\logs\overview.txt"
output_dir = r"d:\C_Drive_Transfer\bron\src\renderer\components"

if not os.path.exists(output_dir):
    os.makedirs(output_dir)

components = [
    "BrowserToolbar.tsx",
    "TabBar.tsx",
    "AgentSidebar.tsx",
    "MemoryPanel.tsx",
    "SettingsPanel.tsx"
]

files_found = {}

with open(log_path, 'r', encoding='utf-8') as f:
    for line in f:
        try:
            data = json.loads(line)
            if "tool_calls" in data:
                for call in data["tool_calls"]:
                    if call["name"] in ["write_to_file", "replace_file_content", "multi_replace_file_content"]:
                        args = call["args"]
                        target = args.get("TargetFile", "")
                        for comp in components:
                            if comp in target:
                                if call["name"] == "write_to_file":
                                    files_found[comp] = args.get("CodeContent", "")
                                # For simplicity, we just take the last write_to_file
                                # We could handle replacements but usually the last write is enough or we can reconstruct
        except:
            continue

for comp, content in files_found.items():
    # Content is double-escaped in the log
    try:
        # First unescape from JSON string
        content = content.strip('"')
        # Then replace escaped newlines and quotes
        content = content.replace('\\n', '\n').replace('\\"', '"').replace('\\\\', '\\')
        
        # Sometimes it's still wrapped in quotes if it was a nested JSON string
        if content.startswith('"') and content.endswith('"'):
            content = content[1:-1].replace('\\n', '\n').replace('\\"', '"')

        dest_path = os.path.join(output_dir, comp)
        with open(dest_path, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Restored {comp}")
    except Exception as e:
        print(f"Error restoring {comp}: {e}")
